release/main.py

```diff
@@ -35,20 +39,69 @@ def tag_all(manifest, tag, delete_if_exists=False):
+    for repo, repo_info in manifest_contnet.items():
+        logger.info("Creating %s tag in repo: %s, revision: %s", tag, repo, repo_info['revision'])
+        gtools.create_tag(repo, repo_info['revision'], tag)
     logger.info("Done")
 
 
+def untag_all(manifest, tag):
+    '''Read the manifest file and delete the tag for each entry'''
+    logger = get_logger()
+    with open(manifest, 'r') as manifest_file:
+        manifest_contnet = yaml.safe_load(manifest_file)
+    logger.info("Deleting %s tag from all repos", tag)
+    gtools = gittools.GitApiUtils()
+    repos_with_tag = []
+    for repo in manifest_contnet.keys():
+        if gtools.tag_exists(repo, tag):
+            repos_with_tag.append(repo)
+    if repos_with_tag:
+        for repo in repos_with_tag:
+            gtools.delete_tag(repo, tag)
+
+
+def check_images_exists(manifest, tag):
+    '''Read the manifest file and check for each repository if all images exists with the given tag
+    Raise exception in case one or more images are missing
+    '''
+    logger = get_logger()
+    with open(manifest, 'r') as manifest_file:
+        manifest_contnet = yaml.safe_load(manifest_file)
+    logger.info("Checking images with %s tag exists for all repos", tag)
+    missing_images = []
+    for repo, repo_info in manifest_contnet.items():
+        for image in repo_info["images"]:
+            if not image_exists(image, tag):
+                logger.warning("Missing image %s from repo: %s", image, repo)
+                missing_images.append(image)
+    if missing_images:
+        raise Exception("Missinge images %s" % missing_images)
+
+
+def image_exists(image_name, tag):
+    '''Return True if image with the given tag exists in quay.io under ocpmetal namespace'''
+    url = IMAGES_URL_TEMPLATE.format(image_name, tag)
+    response = requests.get(url=url)
+    return response.ok
+
+
 def main():
     parser = argparse.ArgumentParser(description='Tag all assisted installer repositories')
     parser.add_argument('-t', '--tag', help='The tag to create', type=str, required=True)
-    parser.add_argument('-f', '--force', help='Delete tag if oreviosly exists', action="store_true")
+    tag_options = parser.add_mutually_exclusive_group()
+    tag_options.add_argument('-d', '--delete', help='Delete the tag from all repos', action="store_true")
+    tag_options.add_argument('-f', '--force', help='Override the tag if previosly exists', action="store_true")
+    tag_options.add_argument('-c', '--check', help='Check if the repository images with the given tag exists', action="store_true")
     parser.add_argument('-m', '--manifest', help='Path to manifest file', type=str, default="./assisted-installer.yaml")
     args = parser.parse_args()
-    tag_all(args.manifest, args.tag, args.force)
-
+    if args.delete:
+      untag_all(args.manifest, args.tag)
+    elif args.check:
+        check_images_exists(args.manifest, args.tag)
+    else:
+        tag_all(args.manifest, args.tag, args.force)
```

Comment:
Maybe also tag the assisted-installer-deployment repo with the same tag?

Reply:
Another PR